File: C++/Robot Return to Origin.cpp
class Solution {
public:
    bool judgeCircle(string moves) {
        
        int ctr = 0;
        int rtc = 0;

        // if (moves.length() % 2 != 0) return false;

        for (char x : moves) {

            switch(x) {
                case 'U':
                    rtc += 1;
                    break;

                case 'D':
                    rtc -= 1;
                    break;

                case 'R':

                    ctr += 1;
                    break;

                case 'L':

                    ctr -= 1;
                    break;

            }

        }

        return (ctr == 0 && rtc == 0);

    }
};

File: python/Robot Return to Origin.py
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        
        ctr = 0
        rtc = 0
        # if len(moves) % 2 != 0:
        #     return False

        for x in moves:
            if x == 'U':
                ctr += 1
            elif x == 'R':
                rtc += 1
            elif x == 'D':
                ctr -= 1
            else:
                rtc -= 1

        return (ctr == 0 and rtc == 0)
